int32 HANDSHAKE_RETURN_CODE_ROBOT_BASE_NOT_ALIVE = 0
int32 HANDSHAKE_RETURN_CODE_ALREADY_GAINED_CONTROL = 1
int32 HANDSHAKE_RETURN_CODE_ALREADY_LOST_CONTROL = 2
int32 HANDSHAKE_RETURN_CODE_CONTROL_ACQUIRED = 3
int32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_ROBOT_BASE_FAULT = 4
int32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_RC_HALT_TRIGGERED = 5
int32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_RC_CONTROL_ACTIVE = 6
int32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_TOKEN_TRANSFER_INTERRUPTED = 7
int32 HANDSHAKE_RETURN_CODE_CONTROL_HANDED_OVER = 8
int32 HANDSHAKE_RETURN_CODE_CONTROL_REQUEST_TIMEOUT = 9
int32 HANDSHAKE_RETURN_CODE_RENOUNCE_REQUEST_TIMEOUT = 10

uint8 REQUEST_CONTROL = 0
uint8 RENOUNCE_CONTROL = 1

uint8 action_type
---
int32 result_code